### ISSUE TITLE
Google deprecating many terms from build.gradle at the end of this year. any plan to upgrade the project?

### ISSUE THREAD

**Issue by Shhzdmrz** [mauron85/react-native-background-geolocation#302]
## Your Environment
RN: 0.57.0
OS: Windows
* Plugin version: 0.5.0-alpha.42
* Platform: Android
* OS version: Windows
* Device manufacturer and model: NOKIA E5
* React Native version: 0.57.0
* Plugin configuration options: N/A 
* Link to your project: N/A

## Context
Its not a bug but its going to be because deprecation of many terms in android at the end of this year.
http://d.android.com/r/tools/update-dependency-configurations.html
## Expected Behavior
Is should not show these warnings.
## Actual Behavior
Showing all these warnings.

## Possible Fix
Upgrade the projects for the terms mentioned in warnings.

## Warnings
_publishNonDefault is deprecated and has no effect anymore. All variants are now published._
> WARNING: Configuration 'androidTestCompile' is obsolete and has been replaced with 'androidTestImplementation' and 'androidTestApi'.
> It will be removed at the end of 2018. For more information see: http://d.android.com/r/tools/update-dependency-configurations.html

> WARNING: Configuration 'androidTestApi' is obsolete and has been replaced with 'androidTestImplementation'.
> It will be removed at the end of 2018. For more information see: http://d.android.com/r/tools/update-dependency-configurations.html

> WARNING: Configuration 'compile' is obsolete and has been replaced with 'implementation' and 'api'.
> It will be removed at the end of 2018. For more information see: http://d.android.com/r/tools/update-dependency-configurations.html

> WARNING: Configuration 'testCompile' is obsolete and has been replaced with 'testImplementation' and 'testApi'.
> It will be removed at the end of 2018. For more information see: http://d.android.com/r/tools/update-dependency-configurations.html

> WARNING: Configuration 'testApi' is obsolete and has been replaced with 'testImplementation'.
> It will be removed at the end of 2018. For more information see: http://d.android.com/r/tools/update-dependency-configurations.htmlspath > protos-25.2.3.jar.sha1

> WARNING: Configuration 'oreoCompile' is obsolete and has been replaced with 'oreoImplementation' and 'oreoApi'.and 'androidTestApi'.
> It will be removed at the end of 2018. For more information see: http://d.android.com/r/tools/update-dependency-configurations.html

**Comment by mauron85** [comment]
Yes, since RN57 we can move to solely to gradle 4, and removing all those warning all together. But this is currently not top priority. If somebody willing to do it, I don't mind merge such a PR.
